oxide/crates/core/src/lib.rs

```diff
@@ -399,6 +400,23 @@ pub fn scan_files(input: Vec<ChangedContent>, options: u8) -> Vec<String> {
     }
 }
 
+fn read_changed_content(c: ChangedContent) -> Option<Vec<u8>> {
+    match (c.file, c.content) {
+        (Some(file), None) => match std::fs::read(&file) {
+            Ok(content) => match file.extension().map(|x| x.to_str()) {
+                Some(Some("svelte")) => Some(content.replace(" class:", " ")),
+                _ => Some(content),
+            },
+            Err(e) => {
+                event!(tracing::Level::ERROR, "Failed to read file: {:?}", e);
+                Default::default()
+            }
+        },
+        (None, Some(content)) => Some(content.into_bytes()),
+        _ => Default::default(),
+    }
+}
```

Comment:
The triple nested match feels a bit weird here. It's _shorter_ but harder to fallow. All but one of these conditions are early-returnable so maybe something like this instead:
```suggestion
fn read_changed_content(c: ChangedContent) -> Option<Vec<u8>> {
  if let Some(content) = c.content {
    return Some(content.into_bytes())
  }

  let Some(file) = c.file else {
    return Default::default()
  }
  
  let Ok(content) = std::fs::read(&file) else {
    event!(tracing::Level::ERROR, "Failed to read file: {:?}", e);
    return Default::default()
  }
  
   match file.extension().map(|x| x.to_str()) {
    Some(Some("svelte")) => Some(content.replace(" class:", " ")),
    _ => Some(content),
  }
}
```

Reply:
I kept the original code and added the extension part to it. Splitting this up is a bit clearer indeed.

I did fix some syntax errors and changed it slightly.